src/Apps/Collect/CollectionApp.tsx

```diff
@@ -13,8 +14,15 @@ interface CollectionAppProps {
 }
 
 export class CollectionApp extends Component<CollectionAppProps> {
+  collectionNotFound = collection => {
+    if (!collection) {
```

Comment:
Also we may want a better check then `collection` because `collection` can be `null` before the page actually loads. We should use the relay request status code